rfcs/0026-anyone-can-pay/0026-anyone-can-pay.md

```diff
@@ -0,0 +1,270 @@
+
+```
+<20 byte blake160 public key hash>
+<20 byte blake160 public key hash> + <1 byte CKByte minimum>
+<20 byte blake160 public key hash> + <1 byte CKByte minimum> + <1 byte UDT minimum>
+```
+
+The additions of CKByte & UDT minimums enforces the minimal amount one can transfer to the anyone-can-pay lock. This provides a mitigation against DDoSing on cell level: if a cell is setup using anyone-can-pay lock, an attacker can keep creating transactions that transfer only 1 shannon or 1 UDT to the cell, make it difficult for the cell owner to claim the tokens stored in the cell. By providing a minimal transfer amount, a user can raise the attacking cost, hence protecting his/her own cells against DDoS attacks. This mechanism won't prevent all kinds of DDoS of course, but it serves as a quick solution to mitigate cheaper ones.
+
+The value stored in CKByte & UDT minimum are interpreted in the following way: if `x` is stored in the field, the minimal transfer amount will be `10^x`, for example:
+
+* If 3 is stored in CKByte minimum, it means the minimal amount that can be accepted by the cell is 1000 shannons
+* If 4 is stored in UDT base unit minimum, it means the minimal amount that can be accepted by the cell is 10000 UDT base units.
+
+Note the minimum fields are completely optional. If a minimum is not provided, we will treat the minimum value as 0, meaning no minimum is enforced on the transfer operation.
+
+## UDT Interpretation
+
+The anyone-can-pay lock assumes that the locked cell follows the [Simple UDT specification](https://talk.nervos.org/t/rfc-simple-udt-draft-spec/4333), thus the cell 1) has a type script; 2) has at least 16 bytes in the cell data part. Its up to the user to ensure one only uses anyone-can-pay lock with a type script implementing Simple UDT specification.
+
+## Unlock Rules
+
+The anyone-can-pay lock will work following the rules below:
+
+1. If a signature is provided, it works exactly as the default secp256k1-blake2b-sighash-all lock, if a signature is provide in witness and can be validated, the lock returns with a success state.
+
+    1.a. If the provided signature fails validation, the lock returns with an error state
+
+2. If a signature is not provided, the lock continues with the added anyone-can-pay logic below:
+
+    2.a. It loops through all input cells using the current anyone-can-pay lock script(notice here the lock script we refer to include public key hash, meaning if a transaction contains 2 cells using the same anyone-can-pay lock code, but different public key hash, they will be treated as different lock script, and each will perform the script unlock rule checking independently), if 2 input cells are using the same type script, or are both missing type scripts, the lock returns with an error state
```

Comment:
Should we mention that different minimums also lead to different lock script?

Reply:
Added in the above structure section